### PR TITLE
Fix dropped tweak when combining signature shares

### PR DESCRIPTION
**Pull request by nickfarrow**
* With the EncodedFrostKey PR we accidentally dropped the tweaks because we use a too-convenient into method.
* This meant we combined the signature shares without the tweak, causing invalid schnorr signatures in our bitcoin wallet.
* We now verify the signature after combining the signature shares.

### CODE REVIEW

File: frostsnap_core/src/coordinator.rs [hunk 1]
```diff
@@ -351,18 +350,25 @@ impl FrostCoordinator {
...
                                 session_progress
                                     .signature_shares
                                     .iter()
                                     .map(|(_, &share)| share)
                                     .collect(),
-                            )
+                            );
+
+                            assert!(session_progress.sign_item.verify(
+                                &Schnorr::verify_only(),
```
Comment (LLFourn) [inline frostsnap_core/src/coordinator.rs:366]: I think we can just use the `frost.schnorr()` instance here.

File: frostsnap_core/src/coordinator.rs [hunk 2]
```diff
@@ -351,18 +350,25 @@ impl FrostCoordinator {
...
+                                session_progress.key.into_frost_key().public_key(),
```
Comment (LLFourn) [inline frostsnap_core/src/coordinator.rs:367]: There's something odd here. Why do we call `session_progress.sign_item.verify` and then pass in another `session_progress` thing into the call. It looks like `.verify` should just be on `session_progress`.

**Comment by LLFourn** [comment]
I pushed a commit addressing my comments and cleaning things up a bit. I think we should change the name of `EncodedFrostKey` to `RootFrostKey` and use it whenever we want a tweakless key.

**Review by LLFourn — APPROVED** [review]

File: frostsnap_core/src/coordinator.rs [hunk 3]
```diff
@@ -670,13 +664,44 @@ pub struct SignSessionProgress {
 }
 
 impl SignSessionProgress {
+    pub fn new<NG>(
+        frost: &Frost<sha2::Sha256, NG>,
+        root_key: EncodedFrostKey,
+        sign_item: SignItem,
+        nonces: BTreeMap<frost::PartyIndex, frost::Nonce>,
+    ) -> Self {
+        // FIXME: This shouldn't be raw -- plain messages should do domain separation
```
Comment (nickfarrow) [inline frostsnap_core/src/coordinator.rs:673]: 👍

will need to look at specs

**Pull request merged**